m3u8/writer.go

```diff
@@ -1069,3 +1334,63 @@ func (p *MediaPlaylist) GetAllSegments() []*MediaSegment {
 /*
 [Protocol Version Compatibility]: https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis-16#section-8
 */
+
+/// Helper functions
+
+func splitUriBy(uri, sep string) (string, string) {
+	// split the uri by the last dot
+	uriParts := strings.Split(uri, sep)
+	if len(uriParts) < 2 {
+		return "", ""
+	}
+	// get the last part of the uri
+	lastPart := uriParts[len(uriParts)-1]
+	// get the rest of the uri
+	rest := strings.Join(uriParts[:len(uriParts)-1], sep)
+	return rest, lastPart
+}
+
+// find the numbers in end of the string
+func getSequenceNum(uriPrefix string) (bool, uint64) {
+	if strings.Contains(uriPrefix, ".") {
+		return false, 0
+	}
+
+	// find the last number in the uri
+	numStr := RegexpNum.FindString(uriPrefix)
+	endWithNum := numStr != "" && strings.HasSuffix(uriPrefix, numStr)
+	if numStr == "" {
+		return endWithNum, 0
+	}
+	num, err := strconv.ParseUint(numStr, 10, 64)
+	if err != nil {
+		return endWithNum, 0
+	}
+	return endWithNum, num
+}
+
+// check if a uri include the other
+func IsPartOf(partialSegUri, segUri string) bool {
```

Comment:
Don't export `IsPartOf`. Also note that doc strings should start with he function name so the doc string should read `// isPartOf checks if partialSegUri matches segUri after removing the file extension`